mobile-experimental/Yote/js/global/utils/calendarUtils.js

```diff
@@ -0,0 +1,223 @@
+      case 0:
+        return "Sunday"
+      case 1:
+        return "Monday"
+      case 2:
+        return "Tuesday"
+      case 3:
+        return "Wednesday"
+      case 4:
+        return "Thursday"
+      case 5:
+        return "Friday"
+      case 6:
+        return "Saturday"
+      default:
+        return "Invalid"
+    }
+  }
+  , getShortDayString(day) {
+    //accepts 0..6 and returns Sunday...Saturday
+    switch(day) {
+      case 0:
+        return "S"
+      case 1:
+        return "M"
+      case 2:
+        return "T"
+      case 3:
+        return "W"
+      case 4:
+        return "T"
+      case 5:
+        return "F"
+      case 6:
+        return "S"
+      default:
+        return "-"
+    }
+  }
+
+  , getMonthString(month) {
+    //accepts 0..11 and returns January...December
+    switch(month) {
+      case 0:
+        return "January"
+      case 1:
+        return "February"
+      case 2:
+        return "March"
+      case 3:
+        return "April"
+      case 4:
+        return "May"
+      case 5:
+        return "June"
+      case 6:
+        return "July"
+      case 7:
+        return "August"
+      case 8:
+        return "September"
+      case 9:
+        return "October"
+      case 10:
+        return "November"
+      case 11:
+        return "December"
+      default:
+        return "Invalid"
+    }
+  }
+
+  , getWeekNum(dayKey) {
+    let mapIndex = parseInt(dayKey) - 1;
+    let weekNum = Math.floor(mapIndex / 7 ) + 1;
+    return weekNum;
+  }
+
+  , getDayNum(dayKey) {
+    let mapIndex = parseInt(dayKey) - 1;
+    let dayNum = mapIndex % 7 + 1;
+    return dayNum;
+  }
+
+  , getDayDate(startDate, dayKey) {
+    var start = calendarUtils.initDateTZ(startDate);
+    var days = parseInt(dayKey) - 1;
+    return start.setDate(start.getDate() + days);
+  }
+
+  , initDateTZ(date = new Date()) {
+    //force local client timezone when determining dates
+    // see web/calendarUtils for implementation notes
+    const msToAdd = new Date().getTimezoneOffset() * 60 * 1000;
+    const intermediate = new Date(date);
+    return new Date(intermediate.getTime() + msToAdd); 
+  }
+
+  //program/date methods
+  , programInInterval(program, start, end) {
+    //helper. returns true if program has any days in specified interval
+    const programStartMS = new Date(program.startDate).getTime();
+    const programEndMS = new Date(program.endDate).getTime();
+    if(start.getTime() <= programStartMS && programStartMS <= end.getTime()) {
+      //program starts in interval
+      return true;
+    } else if(start.getTime() <= programEndMS && programEndMS <= end.getTime()) {
+      //program ends in interval
+      return true;
+    } else if(programStartMS < start.getTime() && programEndMS > end.getTime()) {
+      //program encompasses entire interval
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  //TODO: needs testing.
+  , checkValidProgramDates(newProgram, allPrograms) {
```

Comment:
lol from soflete. I think most of this file might be able to be removed, since we use newer libraries for most of this logic these days